libraries/net/message_oriented_connection.cpp

```diff
@@ -251,6 +255,7 @@ namespace graphene { namespace net {
         }
         ~verify_no_send_in_progress() { var = false; }
       } _verify_no_send_in_progress(_send_message_in_progress);
+      _ready_for_sending->wait();
```

Comment:
`_ready_for_sending` is only cancelled in `destroy_connection`. IMO it is correct then to throw here because the connection is no longer valid anyway.